section3/reflection.md

```diff
@@ -2,12 +2,26 @@
 
 1. In Section 3, we removed even more supports and pushed you to learn a bit more independently, read through technical documentation, and use your resources more. How did that feel?
 
+   It actually felt pretty good. I used Google, YouTube, and Slack as my main resources for things I didn't know. If felt less lonely knowing that people in my cohort were were stumped on similar things.
+
 2. Think of a web or mobile application you use regularly. Where is an example of a place that the developers probably use Arrays? Explain.
 
+   My go-to app is Spotify which seems like a good place to have arrays. My first thought is an artist's song library. It could be called something like `var artistSongs = [...]` and within the array would be all of the songs written by that particular artist. It would allow for any number of elements and could house multiple datatypes if needed.
+
 3. In your own words, explain what an Array method is.
 
-4. What is the purpose of a loop? When we would use a loop in conjunction with array vs. without?
+   An Array method is a built-in function of JavaScript and allows you to carry out a specific job on the Array. If you need to add an element, remove an element, reorder an array, etc. you can use a method to get the job done.
+
+4. What is the purpose of a loop? When would we use a loop in conjunction with array vs. without?
+
+   * The purpose of a loop is to efficiently perform repeated tasks based on a condition. The loop will continue to run until the defined condition returns `false`.
+   * We would use a loop in conjunction with an array when we have a large data set and want run a condition against each element in the data set.
+   * We would use a loop without an array when we want to perform an action any specified number of times.
 
 5. If you had to teach a 5 year old about loops, how would you explain it?
 
+   I would start off by explaining that loops are how we do one task over and over until we are told to stop. A good example of this would be frosting cupcakes. We have 6 cupcakes in a line and we want to frost each one of them until we run out of cupcakes. We start at the first one, squeeze on the frosting, and move to the second one. We frost the second cupcake and then move to the third, and so on and so forth until the sixth cupcake is frosted. Once the sixth cupcake is finished, we stop frosting cupcakes.
+
 6. What questions do you still have about Arrays and loops?
+
+   In the exercise file `loops-arrays.js` we skimmed the surface of what loops can actually do with `if/else` conditions. This is probably a loaded question, but just how involved can they really get? Can we run conditional loops millions of times without breaking the terminal?
```

Comment:
Yes probably is loaded but the TLDR is yes, you can loop over lots and lots of data, with those nested conditions, etc!